### PR TITLE
Makes varediting pneumatic cannon easier and makes the code less snowflake-y

### PR DESCRIPTION
**Pull request by silicons**

### CODE REVIEW

**Comment by optimumtact** [comment]
stop adding vars to shit, it increases compile and access time you dunderheaded buffoon

**Comment by praisenarsie** [comment]
yeah because i want kevinz to spawn super pneumatic cannons on a rev round

File: code/game/objects/items/pneumaticCannon.dm [hunk 1]
```diff
@@ -234,27 +261,12 @@
...
 
-/obj/item/pneumatic_cannon/pie/can_load_item(obj/item/I, mob/user)
-	if(istype(I, /obj/item/reagent_containers/food/snacks/pie))
-		return ..()
-	to_chat(user, "<span class='warning'>[src] only accepts pies!</span>")
-	return FALSE
-
+/obj/item/pneumatic_cannon/pie/Initialize()
+	. = ..()
+	allowed_typecache = typecacheof(/obj/item/reagent_containers/food/snacks/pie)
```
Comment (AnturK) [inline code/game/objects/items/pneumaticCannon.dm]: Make this static for the subtype and assign it.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/objects/items/pneumaticCannon.dm [hunk 2]
```diff
@@ -26,8 +26,31 @@
...
+	. = ..()
+	if(selfcharge)
+		init_charge()
+
+/obj/item/pneumatic_cannon/proc/init_charge()	//wrapper so it can be vv'd easier
+	START_PROCESSING(SSobj, src)
+
+/obj/item/pneumatic_cannon/process()
+	if(++charge_tick >= charge_ticks)
+		fill_with_type(/obj/item/reagent_containers/food/snacks/pie/cream, charge_amount)
```
Comment (AnturK) [inline code/game/objects/items/pneumaticCannon.dm]: Shouldn't this be charge type

**Review by AnturK — CHANGES_REQUESTED** [review]

**Comment by AnturK** [comment]
@kevinz000

**Comment by kevinz000** [comment]
One second

File: code/game/objects/items/pneumaticCannon.dm [hunk 3]
```diff
@@ -75,6 +98,10 @@
 		load_item(IW, user)
 
 /obj/item/pneumatic_cannon/proc/can_load_item(obj/item/I, mob/user)
+	if(!isnull(allowed_typecache) && !allowed_typecache[I.type])
+		if(user)
+			to_chat(user, "<span class='warning'>\The [src] won't fit [I]!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/pneumaticCannon.dm]: Remove \The

Comment (Cyberboss) [reply code/game/objects/items/pneumaticCannon.dm]: check your grammar

File: code/game/objects/items/pneumaticCannon.dm [hunk 4]
```diff
@@ -75,6 +98,10 @@
 		load_item(IW, user)
 
 /obj/item/pneumatic_cannon/proc/can_load_item(obj/item/I, mob/user)
+	if(!isnull(allowed_typecache) && !allowed_typecache[I.type])
```
Comment (Cyberboss) [inline code/game/objects/items/pneumaticCannon.dm]: replace if with is_type_in_typecache(I, allowed_typecache)

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request merged**